packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1527,8 +1551,10 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   @override
   void attach(SemanticsOwner owner) {
     super.attach(owner);
-    assert(!owner._nodes.containsKey(id));
-    owner._nodes[id] = this;
+    assert(_id == null || _id == 0);
+    if (_id == null)
+      _setId(_generateNewId());
```

Comment:
If we want to handle the overflow issue, we have to keep track of currently used ids. Ideally I can keep track of that in constructor and destructor, but that is not supported in dart. I think the best place we can do it will then be in attach and detach.

Reply:
An alternative will be just checking the SemanticsNode._nodes in _generateNewId, it is easier and breaks 0 existing tests, but I found it weird to tie them together.